src/components/QuestDetails/components/Reward/index.tsx

```diff
@@ -4,50 +4,68 @@ import { decimalUnits, parseNumIntoReadableString } from '@hyperplay/utils'
 import classNames from 'classnames'
 
 import { LinkExternal } from '@/assets/images'
+import Button from '@/components/Button'
+import Loading from '@/components/Loading'
 
 import { QuestReward } from '../../types'
 import styles from './index.module.scss'
 
+export interface RewardI18n {
+  claimsLeft: string
+  viewReward: string
+  claimed: string
+  claim: string
+}
+
 export interface RewardProps {
   reward: QuestReward
-  i18n: {
-    claimsLeft: string
-    viewReward: string
-    claimed: string
-  }
+  onClaim: (reward: QuestReward) => void
+  i18n: RewardI18n
 }
 
-export default function Reward({ reward, i18n }: RewardProps) {
+export default function Reward({ reward, i18n, onClaim }: RewardProps) {
   let numClaimsLeftComponent = null
+
   if (reward.numOfClaimsLeft) {
     const formattedNumOfClaimsLeft = parseNumIntoReadableString({
       num: reward.numOfClaimsLeft,
       units: decimalUnits,
-      minValue: '0.0001',
-      maxValue: '999999999'
+      minValue: '0',
+      maxValue: '9999'
```

Comment:
We don't have a lot of room in the layout, I decided to decrease the amount to 4 digits before formatting.